developer_guides/dsp_core/index.rst

```diff
@@ -0,0 +1,81 @@
+.. _dsp_core:
+
+DSP Cores Management
+####################
+
+More and more audio DSPs with multiple cores are implemented in the current
+SoC(System on Chip), to fully ultilize the bandwidth of all cores, and do good
+power saving with power on/off cores dynamically, we need a generic DSP cores
+management with SOF stack.
+
+Context -- pipeline, component, and DSP core
+********************************************
+
+Driver pipeline and widget
+==========================
+
+A pipeline usually contains one scheduler widget (or named 'pipeline widget in
+the FW') + several other widgets, all these widgets are 'swidget's in the
+driver side.
+
+Firmware pipeline, component and DSP core
+=========================================
+
+In the firmware side, each pipeline has a dedicated pipeline task(related to
+pipeline_copy()), to be scheduled on a specified DSP core (set in topology via
+scheduler widget's core).
+
+Each component (normal driver widget) can have its own task also, it can be
+specified to run on a specified DSP core, this is not neccesary to be the same
+one with the pipeline one.
+
+
+How to manage the DSP cores in the driver?
+******************************************
+
+- use a refcount for each DSP core to denote if there is any task ask for
+  using it.
+- increase the refcount via _get() each time a widget (its task actually)
+  ask for it, in widget_setup(), before the widget/component can be created
+  via IPC component_new.
+- decrease the refcount via _put() after a widget is freed, in widget_free().
+- power on a DSP core at the point the refcount is changing from 0 to 1, and
+  power off it at 1 to 0.
+
+How to power on/off a DSP core?
+*******************************
+
+To power on a DSP core, we need to
+
```

Comment:
we first have to explain, that some cores are "host-managed" and some are "DSP-managed."